kong/plugins/acl/handler.lua

```diff
@@ -30,9 +30,15 @@ function ACLHandler:access(conf)
   ACLHandler.super.access(self)
 
   local consumer_id
-  if ngx.ctx.authenticated_credential then
+  if ngx.ctx.authenticated_consumer then
+    consumer_id = ngx.ctx.authenticated_consumer.id
+  end
+
+  if not consumer_id and ngx.ctx.authenticated_credential then
     consumer_id = ngx.ctx.authenticated_credential.consumer_id
-  else
+  end
+
```

Comment:
I'd be in favor of the latter approach ^ though the parens aren't necessary I don't think.